modules/labs-react/ai-assistant-ingress-button/lib/AiAssistantIngressButton.tsx

```diff
@@ -0,0 +1,181 @@
+import {createComponent, focusRing} from '@workday/canvas-kit-react/common';
+import {BaseButton, BaseButtonProps, buttonStencil} from '@workday/canvas-kit-react/button';
+import {createStencil, handleCsProp, calc} from '@workday/canvas-kit-styling';
+import {system} from '@workday/canvas-tokens-web';
+import {CanvasIconTypes} from '@workday/design-assets-types';
+import {systemIconStencil} from '@workday/canvas-kit-react/icon';
+export interface AiAssistantIngressButtonProps
+  extends Omit<BaseButtonProps, 'size' | 'colors' | 'icon' | 'iconPosition' | 'shouldMirrorIcon'> {
+  /**
+   * When true, indicates that the assistant side panel is open.
+   */
+  toggled?: boolean;
+  /**
+   * Used on dark backgrounds.
+   */
+  variant?: 'inverse';
+}
+
+export const aiAssistantIngressButtonStencil = createStencil({
+  extends: buttonStencil,
+  base: {
+    [buttonStencil.vars.background]: system.color.bg.ai.strongest,
+    borderRadius: system.shape.round,
+    height: calc.add(system.space.x10, system.space.x1),
+    width: calc.add(system.space.x10, system.space.x1),
+    transition: 'box-shadow 300ms ease-out, background 300ms ease-out',
+    [systemIconStencil.vars.color]: 'url(#ai-assistant-ingress-button-gradient)',
+    '.wd-icon-ai-assistant-ingress-button': {
+      '.wd-icon-fill': {
+        transition: 'fill 300ms ease-out',
+      },
+      '> linearGradient > stop': {
+        transition: '300ms ease-out',
+        stopColor: system.color.fg.inverse,
+      },
+    },
+
+    ':hover': {
+      [systemIconStencil.vars.color]: 'url(#ai-assistant-ingress-button-gradient)',
+      '.wd-icon-ai-assistant-ingress-button': {
+        'linearGradient > stop:first-child': {
+          stopColor: '#FFA198',
+        },
+        'linearGradient > stop:nth-child(2)': {
+          stopColor: '#FFCAA0',
+        },
+        'linearGradient > stop:nth-child(3)': {
+          stopColor: '#FFCA79',
+        },
+        'linearGradient > stop:nth-child(4)': {
+          stopColor: '#FDCA44',
+        },
+        'linearGradient > stop:nth-child(5)': {
+          stopColor: '#FFB74D',
+        },
```

Comment:
Maybe make these hex colors as variables, i feel it will be easier to migrate to tokens if we decide to have them